### PR TITLE
Test interleaved schedules

### PR DESCRIPTION
**Pull request by kwen2501**
## Description

- Added test_interleave.py
- Added to GPU CI.

## Status
- LoopBFS passes the test.
- Interleaved 1F1B fails with key error re micro-chunk id. (See CI result)

Cc: @H-Huang

### CODE REVIEW

File: test/test_interleave.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    )
+    parser.add_argument(
+        "--chunks",
+        type=int,
+        default=4,
+    )
+    parser.add_argument(
+        "--schedule",
+        type=str,
+        default="dfs",
```
Comment (H-Huang) [inline test/test_interleave.py:133]: update default value

Comment (kwen2501) [reply test/test_interleave.py:133]: Good catch :)

File: test/test_interleave.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    parser.add_argument(
+        "--master_port", type=str, default=os.getenv("MASTER_PORT", "29500")
+    )
+    parser.add_argument(
+        "--cuda", type=int, default=int(torch.cuda.is_available())
+    )
+    parser.add_argument(
+        "--chunks",
+        type=int,
+        default=4,
```
Comment (H-Huang) [inline test/test_interleave.py:128]: added a fix to support this: https://github.com/pytorch/PiPPy/pull/1081

**Review by H-Huang — APPROVED** [review]
Thanks for adding the test!

**Pull request merged**